CommonMark/Formatters/HtmlFormatterSlim.cs

```diff
@@ -30,6 +31,8 @@ internal static class HtmlFormatterSlim
             true,  true,  true,  true,  true,  true,  true,  true,  true,  true,  true,  false, false, false, false, false
         };
 
+        private static readonly Dictionary<string, uint> GitHubHeaderIdCounts = new Dictionary<string, uint>();
```

Comment:
If you insist that this class be a static class, how can you make this thread-safe? Only way I can think of is to put a lock around the whole `BlocksToHtmlInner` method, but that seems like it would create a bottleneck.